osm2gtfs/tests/creators/creators_tests.py

```diff
@@ -0,0 +1,368 @@
+        routes_cache_file = os.path.join(
+            self.standard_values['data_dir'], self.selector + "-routes.pkl")
+        stops_file = os.path.join(
+            self.standard_values['data_dir'], self.selector + "-stops.pkl")
+        self.assertTrue(
+            os.path.isfile(stops_file), "The stops cache file doesn't exists")
+        self.assertTrue(
+            os.path.isfile(routes_cache_file), "The routes cache file doesn't exists")
+
+        data = OsmConnector(self.config)
+
+        # Define (transitfeed) schedule object for GTFS creation
+        feed = transitfeed.Schedule()
+        # Initiate creators for GTFS components through an object factory
+        factory = CreatorFactory(self.config)
+        agency_creator = factory.get_agency_creator()
+        feed_info_creator = factory.get_feed_info_creator()
+        routes_creator = factory.get_routes_creator()
+        stops_creator = factory.get_stops_creator()
+        schedule_creator = factory.get_schedule_creator()
+        trips_creator = factory.get_trips_creator()
+
+        # Add data to schedule object
+        agency_creator.add_agency_to_feed(feed)
+        feed_info_creator.add_feed_info_to_feed(feed)
+        stops_creator.add_stops_to_feed(feed, data)
+        routes_creator.add_routes_to_feed(feed, data)
+        schedule_creator.add_schedule_to_data(data)
+        trips_creator.add_trips_to_feed(feed, data)
+
+        # Write GTFS
+        feed.WriteGoogleTransitFeed(self.config.output)
+        gtfs_expected_result = self.standard_values['reference_gtfs']
+        gtfs_generated_result = self.standard_values['output_file']
+
+        # Verify GTFS
+        self.assertTrue(
+            CreatorsTestsHelper.is_valid_gtfs(
+                gtfs_generated_result, self.required_values['gtfs_files']),
+            'The generated GTFS is not valid')
+
+        self.assertTrue(
+            CreatorsTestsHelper.is_valid_gtfs(
+                gtfs_expected_result, self.required_values['gtfs_files']),
+            'The expected GTFS is not valid')
+
+        self.assertTrue(
+            CreatorsTestsHelper.is_identical_gtfs(gtfs_expected_result, gtfs_generated_result),
+            "The generated GTFS is different from the expected one")
+
+        # Obtain and print basic count infos about the GTFS
+        gtfs_infos = CreatorsTestsHelper.get_gtfs_infos(gtfs_generated_result)
+        print("Information about the GTFS feed: " + str(gtfs_infos))
+
+        # Verify stop and route counts against expected values
+        self.assertEqual(gtfs_infos["stop_points_count"], self.required_values['stops_count'],
+                         "Wrong stop_points count in the generated GTFS")
+        self.assertEqual(gtfs_infos["stop_areas_count"], self.required_values['stations_count'],
+                         "Wrong stop_areas count in the generated GTFS")
+        self.assertEqual(gtfs_infos["routes_count"], self.required_values['routes_count'],
```

Comment:
Can we switch the order of the tests ?
The tests about the required values provide much more useful info to debug and to inspect the potential differences than the `is_identical_gtfs` one.

Reply:
Of course, but probably better to do in a separate PR.